torch_geometric/nn/models/label_usage.py

```diff
@@ -0,0 +1,120 @@
+    ):
+
+        super().__init__()
+        self.base_model = base_model
+        self.num_classes = num_classes
+        self.split_ratio = split_ratio
+        self.num_recycling_iterations = num_recycling_iterations
+        self.return_tuple = return_tuple
+        self.training = training
+
+    def forward(
+        self,
+        feat: Tensor,
+        edge_index: Adj,
+        y: Tensor,
+        mask: Tensor,
+    ):
+        r"""Forward pass using label usage algorithm.
+
+        Args:
+            feat (torch.Tensor): Node feature tensor of dimension (N,F)
+                where N is the number of nodes and
+                F is the number of features per node
+            edge_index (torch.Tensor or SparseTensor): The edge connectivity
+                to be passed to base_model
+            y (torch.Tensor): Node ground-truth labels tensor of dimension
+                of (N,) for 1D tensor or (N,1) for 2D tensor
+            mask (torch.Tensor): A mask or index tensor denoting which nodes
+                are used during training
+        """
+        assert feat.dim() == 2, f"feat must be 2D but got shape {feat.shape}"
+        assert y.dim() == 1 or (y.dim() == 2 and y.size(1) == 1),\
+            f"Expected y to be either (N,) or (N, 1), but got shape {y.shape}"
+
+        # set unlabeled mask for unlabeled indices
+        unlabeled_mask = torch.ones(feat.size(0), dtype=torch.bool).to(feat.device)
+
+        # add labels to features for train_labels nodes if in training
+        # else fill true labels for all nodes in mask
+        # zero value nodes in train_pred
+        onehot = torch.zeros([feat.shape[0], self.num_classes]).to(feat.device)
+        if self.training:
+            # random split mask based on split ratio
+            if mask.dtype == torch.bool:
+                split_mask = torch.rand(mask.sum()) < self.split_ratio
+                train_labels = mask.nonzero(as_tuple=False).view(-1)[split_mask] # convert to index tensor
+                train_pred = mask.nonzero(as_tuple=False).view(-1)[~split_mask] 
+            else:
+                split_mask = torch.rand(mask.shape) < self.split_ratio
+                train_labels = mask[split_mask] # D_L: nodes with labels
+                train_pred = mask[~split_mask] # D_U: nodes to predict labels
+
+            unlabeled_mask[train_labels] = False
+
+            # create a one-hot encoding according to tensor dim
+            if y.dim() == 2:
+                onehot[train_labels, y[train_labels, 0]] = 1
+            else:
+                onehot[train_labels, y[train_labels]] = 1
+        else:
+            unlabeled_mask[mask] = False
+
+            # create a one-hot encoding according to tensor dim
+            if y.dim() == 2:
+                onehot[mask, y[mask, 0]] = 1
+            else:
+                onehot[mask, y[mask]] = 1
+                
+        feat = torch.cat([feat, onehot], dim=-1)
+
+        # label reuse procedure
```

Comment:
nit: move this comment down one line.  Line 110 is "label usage" but not "label reuse"